packages/ethereum/contracts/test/Crypto.t.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.6.0 <0.9.0;
+
+import 'forge-std/Test.sol';
+import './utils/Accounts.sol';
+import 'solcrypto/SECP2561k.sol';
+
+import '../src/Crypto.sol';
+
+contract CryptoProxy is HoprCrypto {
+}
+
+contract Crypto is Test, AccountsFixtureTest, HoprCrypto {
+  struct CurvePoint {
+    uint256 x;
+    uint256 y;
+  }
+
+  SECP2561k secp256k1;
+  CryptoProxy crypto;
+  function setUp() public {
+    // use dummy proxy to per-method have gas measurements
+    crypto = new CryptoProxy();
+    secp256k1 = new SECP2561k();
+  }
+
+  function testPointToAddress() public {
+    (bool success, bytes memory returnValue) = address(this).staticcall(abi.encodeWithSelector(HoprCrypto.pointToAddress.selector, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+
+    assertTrue(success);
+    assertEq(address(uint160(uint256(bytes32(returnValue)))), accountA.accountAddr);
+  }
+
+  function testIsCurvePoint() public { 
+    assertTrue(crypto.isCurvePointInternal(0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+  }
+
+  function testRevert_NoCurvePoint() public {
+    assertFalse(crypto.isCurvePointInternal(0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75));
+  }
+
+  function testScalarTimeBasepoint() public {
+    assertEq(crypto.scalarTimesBasepoint(accountA.privateKey), accountA.accountAddr);
+  }
+
+  function testEcAddPointAddition() public {
```

Comment:
Can we also test other failure modes ? ( P = Q, P + non-curve point,...)
I see below only a test for P + (-P) in `testRevert_EcAddEdgeCase`

Reply:
The question is here is how often to do this check since the check is not for free (in terms of execution cost). I prefer to do it in the redeemTicket function in HoprChannels smart contract and change the method description to sth saying that it's the duty of the caller to make sur it's a curve point.